wp-convertkit.php

```diff
@@ -281,48 +341,81 @@ private static function _get_settings_page_link($query_args = array()) {
 			return add_query_arg($query_args, admin_url('options-general.php'));
 		}
 
+
 		/**
-		 * Retrieve hosted form markup form the API and format
-		 *
-		 * @param $attributes
-		 * @return string
+		 * Run version specific upgrade.
 		 */
-		public static function get_form_embed($attributes) {
-			$attributes = shortcode_atts(array(
-				'form' => -1,
-			), $attributes);
+		public static function upgrade() {
 
-			$form = $attributes['form'];
+			$current_version = get_option( 'convertkit_version' );
 
-			$form_id = intval(($form < 0) ? self::_get_settings('default_form') : $form);
-			$form = false;
+			if ( ! $current_version) {
 
-			if ($form_id == 0) {
-				return "";
-			}
+				// Run 1.4.1 upgrade
+				$settings = self::_get_settings( );
 
-			$forms_available = self::$api->get_resources('forms');
-			foreach($forms_available as $form_available) {
-				if($form_available['id'] == $form_id) {
-					$form = $form_available;
-					break;
-				}
-			}
+				if ( isset( $settings['api_key'] ) ) {
 
-			$url = add_query_arg( array(
-					'api_key' => self::_get_settings('api_key'),
-					'v' => self::$forms_version,
-					),
-				'https://forms.convertkit.com/' . $form['id'] . '.html'
-			);
+					// Get all posts and pages to track what has been updated
+					$posts = get_option( '_wp_convertkit_upgrade_posts' );
+
+					if ( ! $posts ) {
+
+						$args = array(
+							'post_type' => array( 'post', 'page' ),
+							'fields' => 'ids',
+						);
+
+						$result = new WP_Query( $args );
```

Comment:
I'm conspicuously only getting 10 results to loop through. Does this need a limit set/unset?